### ISSUE TITLE
Convert old Qt signal connection scheme to the typesafe new one

### ISSUE THREAD

**Issue by kintel** [openscad/openscad#5547]
Qt5 introduced a more modern way of connecting signals to functions: https://wiki.qt.io/New_Signal_Slot_Syntax

The new way offers the benefit of being able to perform checks at compile-time for whether the signal and slot exists and that their parameters are correct.

TODO: Convert all manual `connect()` invocations to use the new style.

**Comment by akashmahesh2805** [comment]
Hi, this seems like a good first issue for me to start contributing! Could you please assign it to me?

**Comment by scat25** [comment]
Hi, I would like to fix this issue. The existing pr is stalled, is it ok if I create a new pr for this?

**Comment by t-paul** [comment]
@scat25 Sure, you can create a new PR. We can clean-up open PRs once something is merged.
